### PR TITLE
redpanda: move statefulset.yaml to go

### PR DESCRIPTION
**Pull request by jan-g**

### CODE REVIEW

**Comment by jan-g** [comment]
Needs (a) tidying up the cruft; (b) regenerate the test values; (c) investigate the source of the differences between the config checksum values in the old and the new.

File: charts/redpanda/statefulset.go [hunk 1]
```diff
@@ -847,3 +886,442 @@ func templateToContainers(dot *helmette.Dot, template string) []corev1.Container
...
+					PriorityClassName:             values.Statefulset.PriorityClassName,
+					Tolerations:                   statefulSetTolerations(dot),
+				},
+			},
+			VolumeClaimTemplates: nil, //TODO
+		},
+	}
+
+	// TopologySpreadConstraints
+	// Was protected with this: semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion
```
Comment (chrisseto) [inline charts/redpanda/statefulset.go:940]: I think it's fairly safe to drop support for 1.16. Turns out the minimum required version defined by Chart.yaml is 1.21. So very safe to drop actually.

File: pkg/gotohelm/helmette/sprig.go [hunk 2]
```diff
@@ -400,3 +402,9 @@ func Join[T any](sep string, s []T) string {
 	}
 	return out
 }
+
+// +gotohelm:builtin=sha256sum
+func Sha256Sum(input string) string {
+	hash := sha256.Sum256([]byte(input))
+	return hex.EncodeToString(hash[:])
```
Comment (chrisseto) [inline pkg/gotohelm/helmette/sprig.go:409]: Remind me the purpose of `[:]`? I was under the impression that it's functionally a no-op or there's something tricky about strings vs slices vs arrays.

File: charts/redpanda/statefulset.go (hunk 1 above)
Comment (jan-g) [reply charts/redpanda/statefulset.go]: In which case I'l drop the "Capabilities" bit :-)

File: pkg/gotohelm/helmette/sprig.go (hunk 2 above)
Comment (jan-g) [reply pkg/gotohelm/helmette/sprig.go:409]: Sum256 returns a fixed-length array; EncodeToSting wants a slice.

File: charts/redpanda/statefulset.go (hunk 1 above)
Comment (jan-g) [reply charts/redpanda/statefulset.go]: (done)

**Review by chrisseto — APPROVED** [review]

**Pull request merged**